File: Java Programs/2.2.1 TicTacToe.java
/*
# - - - - - - - - - - - - - - - - #
# SummerTech/Coditum Curriculum   #
# Created by Gabe Ehrlich         #
#                                 #
# Report any bugs, issues, or     #
# corrections to:                 #
# matt@summertech.net             #
#                                 #
#                                 #
# Module 2.2.1                    #
# - - - - - - - - - - - - - - - - #

SUMMERTECH CONFIDENTIAL
COPYRIGHT (C) 2017 SUMMERTECH, INC. ALL RIGHTS RESERVED
*/

import java.util.Scanner;
public class TicTacToe {
  
  public static void main(String[] args) {
    
    //Variable setup
    Scanner scan=new Scanner(System.in);
    char turn='X';
    char[][] board=new char[3][3];
    int count=0;
    
    //Set the board
    for(int i=0;i<3;i++) {
      for(int j=0;j<3;j++) {
        board[i][j]='-';
      }
    }
    
    while(count < 9) {
      //Print the board
      for(int i=0;i<3;i++) {
        for(int j=0;j<3;j++) {
          System.out.print(board[j][i]+" ");
        }
        System.out.println();
      }
      
      //Gameplay
      System.out.println("Player "+ turn +" what x coordinate would you like to play");
      int xcoord = scan.nextInt();
      System.out.println("Player "+ turn +" what y coordinate would you like to play");
      int ycoord = scan.nextInt();
      
      if(xcoord >= 0 && xcoord < 3 && ycoord >= 0 && ycoord < 3) {
        if(board[xcoord][ycoord] == '-') {
          board[xcoord][ycoord] = turn;
          if(turn == 'X')
            turn = 'O';
          else if(turn == 'O')
            turn = 'X';
          count++;
        }
        else {
          System.out.println("Invalid move.");
        }
      }
      else {
        System.out.println("Invalid move.");
      }
    }
    
    //Win conditions
    if(board[0][0] == turn && board[0][1] == turn && board[0][2] == turn) {
      System.out.println(turn + " wins!");
      System.exit(0);
    }
    else if(board[1][0] == turn && board[1][1] == turn && board[1][2] == turn) {
      System.out.println(turn + " wins!");
      System.exit(0);
    }
    else if(board[2][0] == turn && board[2][1] == turn && board[2][2] == turn) {
      System.out.println(turn + " wins!");
      System.exit(0);
    }
    else if(board[0][0] == turn && board[1][0] == turn && board[2][0] == turn) {
      System.out.println(turn + " wins!");
      System.exit(0);
    }
    else if(board[0][1] == turn && board[1][1] == turn && board[2][1] == turn) {
      System.out.println(turn + " wins!");
      System.exit(0);
    }
    else if(board[0][2] == turn && board[1][2] == turn && board[2][2] == turn) {
      System.out.println(turn + " wins!");
      System.exit(0);
    }
    else if(board[0][0] == turn && board[1][1] == turn && board[2][2] == turn) {
      System.out.println(turn + " wins!");
      System.exit(0);
    }
    else if(board[2][0] == turn && board[1][1] == turn && board[0][2] == turn) {
      System.out.println(turn + " wins!");
      System.exit(0);
    }
    
    //Nine turns have gone and the game hasn't ended, must be a tie!
    System.out.println("It's a tie!");
  }
  
}

File: Python Programs/2.2.1 TicTacToe.py
# - - - - - - - - - - - - - - - - #
# SummerTech/Coditum Curriculum   #
# Created by Matthew Baptist      #
#                                 #
# Report any bugs, issues, or     #
# corrections to:                 #
# matt@summertech.net             #
#                                 #
#                                 #
# Module 2.2.1                    #
# - - - - - - - - - - - - - - - - #

#SUMMERTECH CONFIDENTIAL
#COPYRIGHT (C) 2017 SUMMERTECH, INC. ALL RIGHTS RESERVED

board = []

for i in range(3):
  temp = []
  for j in range(3):
    temp.append(" ")
  board.append(temp)

count = 0
letter = "x"

while count < 9 and count >= 0:

  print(" " + board[0][0] + " | " + board[0][1] + " | " + board[0][2] )
  print(" --------- ")
  print(" " + board[1][0] + " | " + board[1][1] + " | " + board[1][2] )
  print(" --------- ")
  print(" " + board[2][0] + " | " + board[2][1] + " | " + board[2][2] )


  x = int(input("Please enter the x-coordinate of your piece. "))
  y = int(input("Please enter the y-coordinate of your piece. "))

  board[y][x] = letter
  
  count += 1

  if board[0][0] == letter and board[0][1] == letter and board[0][2] == letter:
    print(letter, "wins!")
    count = -1
  
  if board[1][0] == letter and board[1][1] == letter and board[1][2] == letter:
    print(letter, "wins!")
    count = -1

  if board[2][0] == letter and board[2][1] == letter and board[2][2] == letter:
    print(letter, "wins!")
    count = -1

  if board[0][0] == letter and board[1][0] == letter and board[2][0] == letter:
    print(letter, "wins!")
    count = -1

  if board[0][1] == letter and board[1][1] == letter and board[2][1] == letter:
    print(letter, "wins!")
    count = -1

  if board[0][2] == letter and board[1][2] == letter and board[2][2] == letter:
    print(letter, "wins!")
    count = -1

  if board[0][0] == letter and board[1][1] == letter and board[2][2] == letter:
    print(letter, "wins!")
    count = -1

  if board[2][0] == letter and board[1][1] == letter and board[0][2] == letter:
    print(letter, "wins!")
    count = -1


  if letter == "x":
    letter = "o"
  else:
    letter = "x"

if count == 9:
  print("Tie game!")
